src/Swift.Runtime/src/Swift/Runtime/ProtocolWitnessTable.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+
+namespace Swift.Runtime;
+
+public readonly struct ProtocolConformanceDescriptor : IEquatable<ProtocolConformanceDescriptor>
+{
+    private readonly IntPtr _handle;
+
+    private ProtocolConformanceDescriptor(IntPtr handle)
+    {
+        _handle = handle;
+    }
+
+    public readonly static ProtocolConformanceDescriptor Zero = default;
+
+    public bool IsValid => _handle != IntPtr.Zero;
+
+    public bool Equals(ProtocolConformanceDescriptor other)
+    {
+        return _handle == other._handle;
+    }
+
+    internal static bool TryGetProtocolConformanceDescriptor<T, U>([NotNullWhen(true)] out ProtocolConformanceDescriptor? result)
```

Comment:
Naming - I would use more descriptive names for the generic type parameters `TType`, `TProtocol` for example.